ui/nuxeo-page-provider-display-behavior.js

```diff
@@ -240,11 +250,18 @@ export const PageProviderDisplayBehavior = [
     _updateResults() {
       if (this._hasPageProvider()) {
         this.size = this.items.length;
+        // because the items might not have been initialized, we use the resultsCount to control the display
+        this._resultsCount = this.nxProvider.resultsCount;
       }
     },
 
     _itemsChanged() {
       this._isEmpty = !(this.items && this.items.length > 0);
+      // guarantee that the size and results count are both updated when the items change
+      this._resultsCountDebouncer = Debouncer.debounce(this._resultsCountDebouncer, timeOut.after(100), () => {
```

Comment:
It seems that when we set the `items` directly in the table, instead of using a page provider to get them, we are not updating the size `property`. I suggest we execute the following code to fix this and the original problem. Wdyt @Gabez0r? This solves the problem of having an empty array of items too (if you have the time, can you try this locally too? Thanks).

Reply:
Thanks Ricardo. Does this aim at fixing the "Loading" issue? I still see it on the [preview](https://preview-webui-fix-elements-1383-search-results-empty-label.webui.dev.nuxeo.com/?path=/story/ui-nuxeo-data-table--empty).

Reply:
🤔 For some reason the preview is not updated. I am running the GH again to see if it is updated (if you try it locally the `Loading` issue is fixed)

Reply:
Indeed. Tested locally, but now I see a flashing of "Loading" before getting an "Empty results". Was this intended?

Reply:
It looks much better, though!

Reply:
No, it was not intended in fact, but it kind of makes sense no? Otherwise, we end up with the opposite behavior, flashing a `No results` before the `Loading` when we are actually fetching the results from the page provider.

Reply:
It's just that displaying Loading when we are not loading anything is misleading.

Reply:
Correct, you are right 😕

Reply:
I feel that we are replacing a flash that we didn't want with another that we might not want as well... 🤔

Reply:
What about if look at the problem fro ma different angle. Perhaps we are just displaying the label too soon, and simply showing a bit later won't make any difference for the use and will prevent the flashing. Something like this (diff from `master`):
```diff
diff --git a/ui/nuxeo-page-provider-display-behavior.js b/ui/nuxeo-page-provider-display-behavior.js
index 50bc259f..3021893a 100644
--- a/ui/nuxeo-page-provider-display-behavior.js
+++ b/ui/nuxeo-page-provider-display-behavior.js
@@ -47,11 +47,6 @@ export const PageProviderDisplayBehavior = [
         type: String,
       },
 
-      _computedEmptyLabel: {
-        type: String,
-        computed: '_computeLabel(i18n, emptyLabel, filters, loading)',
-      },
-
       _isEmpty: {
         type: Boolean,
         value: true,
@@ -194,6 +189,7 @@ export const PageProviderDisplayBehavior = [
       '_nxProviderChanged(nxProvider)',
       '_selectionEnabledChanged(selectionEnabled, selectOnTap)',
       '_itemsChanged(items.*)',
+      '_computeLabel(i18n, emptyLabel, filters, loading)',
     ],
 
     listeners: {
@@ -447,13 +443,16 @@ export const PageProviderDisplayBehavior = [
     },
 
     _computeLabel() {
-      if (this.loading) {
-        return this.i18n('label.loading');
-      }
-      if (this.filters && this.filters.length > 0) {
-        return this.emptyLabelWhenFiltered ? this.emptyLabelWhenFiltered : this.i18n('label.noResultsWhenFiltered');
-      }
-      return this.emptyLabel ? this.emptyLabel : this.i18n('label.noResults');
+      this._computeLabelDebouncer = Debouncer.debounce(this._computeLabelDebouncer, timeOut.after(500), () => {
+        if (this.loading) {
+          this._computedEmptyLabel = this.i18n('label.loading');
+        } else if (this.filters && this.filters.length > 0) {
+          this._computedEmptyLabel = this.emptyLabelWhenFiltered ?
+            this.emptyLabelWhenFiltered : this.i18n('label.noResultsWhenFiltered');
+        } else {
+          this._computedEmptyLabel =  this.emptyLabel ? this.emptyLabel : this.i18n('label.noResults');
+        }
+      });
     },
 
     _quickFilterChanged() {

```